lib/galaxy/tools/parameters/validation.py

```diff
@@ -248,6 +249,23 @@ def validate( self, value, trans=None ):
                 raise ValueError( self.message )
 
 
+class DatasetFilesPathEmptyValidator( Validator ):
+    """Validator that checks if a dataset's files_path exists and is not empty."""
+    def __init__( self, message=None ):
+        self.message = message
+
+    @classmethod
+    def from_element( cls, param, elem ):
+        return cls( elem.get( 'message', None ) )
+
+    def validate( self, value, trans=None ):
+        if value:
+            if not os.path.isdir(value.extra_files_path) or len(os.listdir(value.extra_files_path)) == 0:
```

Comment:
The idea is that total_size is the sum of primary file size and all files in extra_files_path or dataset_%d_files. So the extra files total size is zero if value.get_total_size() == value.get_size(). It does call self.object_store.exists with the extra_files_path, so that should be safe.